src/dmd/expressionsem.d

```diff
@@ -9675,13 +9694,14 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 tsa2 = cast(TypeSArray)toStaticArrayType(se);
             else
                 tsa2 = t2.isTypeSArray();
+
             if (tsa1 && tsa2)
             {
                 uinteger_t dim1 = tsa1.dim.toInteger();
                 uinteger_t dim2 = tsa2.dim.toInteger();
                 if (dim1 != dim2)
                 {
-                    exp.error("mismatched array lengths, %d and %d", cast(int)dim1, cast(int)dim2);
```

Comment:
Static arrays can be really large, I don't think it's a good idea to print the entire expression on the command line. The error location already points to it.

It's not part of the bug fix either.

Reply:
I agree with @dkorpel on this one. Either way, modifying this error message should the the subject of a different PR.

Reply:
We write out the expression in many error messages. Besides, I argued for decades that assert doesn't need to print the expression, just the file/line. Nobody agrees with me, because now it prints the expression.

> Static arrays can be really large

It prints the expression, not the array contents.

> It's not part of the bug fix either.

It's in the same section of code. Improving one line when the opportunity presents itself is not a big deal.

Reply:
Fair enough.